variable/test/transform_test.cpp

```diff
@@ -772,9 +772,10 @@ TEST_F(TransformInPlaceBucketsDryRunTest, unchanged_if_success) {
 TEST(TransformFlagsTest, no_variance_on_arg) {
   auto var_with_variance = makeVariable<double>(Values{1}, Variances{1});
   auto var_no_variance = makeVariable<double>(Values{1});
-  auto binary_op = [](auto x, auto y) { return x + y; };
-  auto op_arg_0_has_flags =
+  constexpr auto binary_op = [](auto x, auto y) { return x + y; };
```

Comment:
Looks like this relates to the error that those variables were used uninitialised. Do you know why `constexpr` fixes it?

Reply:
In all honesty, no, I'm not sure why adding `constexpr` fixes it. From https://docs.microsoft.com/en-us/cpp/cpp/lambda-expressions-constexpr?view=msvc-160 I believe this lambda would have been "implicitly constexpr" anyway. The line below with the `scipp::overloaded` call (also changed to `constexpr`) is where it was actually triggering the breakpoint (not sure how accurate this location really is). Everything looks like it was initialized before being used to me... and if it wasn't, it shouldn't compile with constexpr... unless there is some subtelty I've missed with `scipp::overloaded` or one of the `transform_flags`.

The other uses of `scipp::overloaded` in scipp to all use either `constexpr` or use/return the result immediately.

Reply:
`overloaded` only cares about the type, it does not use the instance of the lambda you pass in at all. Maybe it has to do with how the `overloaded` objects' parents are constructed.

But as you say, if it were truly uninitialised, it shouldn't compile.

I will just keep in mind to make all transform kernel `constexpr`